packages/flutter/lib/src/material/ink_decoration.dart

```diff
@@ -0,0 +1,356 @@
+///   ),
+/// )
+/// ```
+///
+/// The following example shows how an image can be printed on a [Material]
+/// widget with an [InkWell] above it:
+///
+/// ```dart
+/// new Material(
+///   color: Colors.grey[800],
+///   child: new Center(
+///     child: new Ink.image(
+///       image: new AssetImage('cat.jpeg'),
+///       fit: BoxFit.cover,
+///       width: 300.0,
+///       height: 200.0,
+///       child: new InkWell(
+///         onTap: () { /* ... */ },
+///         child: new Align(
+///           alignment: Alignment.topLeft,
+///           child: new Padding(
+///             padding: const EdgeInsets.all(10.0),
+///             child: new Text('KITTEN', style: new TextStyle(fontWeight: FontWeight.w900, color: Colors.white)),
+///           ),
+///         )
+///       ),
+///     ),
+///   ),
+/// )
+/// ```
+///
+/// See also:
+///
+///  * [Container], a more generic form of this widget which paints itself,
+///    rather that defering to the nearest [Material] widget.
+///  * [InkDecoration], the [InkFeature] subclass used by this widget to paint
+///    on [Material] widgets.
+///  * [InkWell] and [InkResponse], which also draw on [Material] widgets.
+class Ink extends StatefulWidget {
+  /// Paints a decoration (which can be a simple color) on a [Material].
+  ///
+  /// The [height] and [width] values include the [padding].
+  ///
+  /// The `color` argument is a shorthand for `decoration: new
+  /// BoxDecoration(color: color)`, which means you cannot supply both a `color`
+  /// and a `decoration` argument. If you want to have both a `color` and a
+  /// `decoration`, you can pass the color as the `color` argument to the
+  /// `BoxDecoration`.
+  Ink({
+    Key key,
+    this.padding,
+    Color color,
+    Decoration decoration,
+    this.width,
+    this.height,
+    this.child,
+  }) : assert(padding == null || padding.isNonNegative),
+       assert(decoration == null || decoration.debugAssertIsValid()),
+       assert(color == null || decoration == null,
+         'Cannot provide both a color and a decoration\n'
+         'The color argument is just a shorthand for "decoration: new BoxDecoration(color: color)".'
+       ),
+       decoration = decoration ?? (color != null ? new BoxDecoration(color: color) : null),
+       super(key: key);
+
+  /// Creates a widget that shows an image (obtained from an [ImageProvider]) on
+  /// a [Material].
+  ///
+  /// This argument is a shorthand for passing a [BoxDecoration] that has only
+  /// its [BoxDecoration.image] property set to the [new Ink] constructor. The
+  /// properties of the [DecorationImage] of that [BoxDecoration] are set
+  /// according to the arguments passed to this method.
+  ///
+  /// The `image` argument must not be null. The `alignment`, `repeat`, and
+  /// `matchTextDirection` arguments must not be null either, but they have
+  /// default values.
+  ///
+  /// See [paintImage] for a description of the meaning of these arguments.
+  Ink.image({
```

Comment:
I don't think we want `DecoratedBox.image` and `Container.image`.

If we had `BoxDecoration.image` then one could just write new `Ink(decoration: new BoxDecoration.image(...))`

Reply:
Why don't you think we want DecoratedBox.image and Container.image? I mean, I don't know if we do, but maybe we do, if this works out well here...

The problem with only having BoxDecoration.image is that it's still pretty verbose, c.f.:

```dart
  new Ink.image(
    image: new AssetImage('cat.jpeg'),
    fit: BoxFit.cover,
    width: 300.0,
    height: 200.0,
 )

  new Ink(
    decoration: new BoxDecoration.image(
      image: new AssetImage('cat.jpeg'),
      fit: BoxFit.cover,
    ),
    width: 300.0,
    height: 200.0,
 )
```

Reply:
IMHO we'd want `DecoratedBox.image` and `Container.image` constructors for the sake symmetry (because that makes the APIs easier to understand) and we wouldn't want `Ink.image`, et al because less API is more.

IMHO `BoxDecoration.image` balances the API weight than the other 3 constructors, but I defer to you.

Reply:
I think we might want DecoratedBox.image and Container.image, I just think we should see how Ink.image goes before doing it. :-)

Reply:
[video_player](https://pub.flutter-io.cn/packages/video_player) provide a `Texture`, not a `Image`. 
Do you have a solution?